clusterloader2/pkg/ticker/ticker.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ticker
```

Comment:
Golang timer would still need to be wrapped in some addition structure. Tuning set can have burst of qps which I think cannot be expressed by golang timer.
Plus I can keep Ticker simple with Timer being it's strategy.

Reply:
Also note that there's golang [ticker](https://golang.org/pkg/time/#Ticker) which seems to be doing basically what you've implemented below.

Reply:
> I'm still not sure why you need sth more than golang timer. For burst, which is defined as 'a load that generates a burst of a given size every X seconds', you can still use the timer with frequency of X seconds (you'll just need to make N calls every X seconds).

If I push whole logic of tuning set to the ticker, then I agree that it can be done.
If we separate logic from ticker to different objects then I think we will have the same design as here with Sleep being replaced with golang timer. It is impossible to hide golang timer behind an interface - timer channel can be obtained only by accessing struct field, so you need some kind of wrapper anyway.